bindings/CMakeLists.txt

```diff
@@ -28,55 +28,43 @@
 # Boston, MA 02111-1307, USA.
 # =============================================================================
 
-set (PY_BINDING_SRCS
-  ${CMAKE_CURRENT_SOURCE_DIR}/bind_py_module.cc
-  ${CMAKE_CURRENT_SOURCE_DIR}/bind_py_structure_manager.cc
-  ${CMAKE_CURRENT_SOURCE_DIR}/bind_py_utils.cc
-  ${CMAKE_CURRENT_SOURCE_DIR}/bind_py_math.cc
-  ${CMAKE_CURRENT_SOURCE_DIR}/bind_py_representation_calculator.cc
-  ${CMAKE_CURRENT_SOURCE_DIR}/bind_py_models.cc
-  )
-
+set(BINDING_SRCS
+    bind_py_module.cc
+    bind_py_structure_manager.cc
+    bind_py_utils.cc
+    bind_py_representation_calculator.cc
+    bind_py_models.cc
+)
 
 # copy rascal folder in the python distribution
 install(DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}/rascal" DESTINATION "${PYTHON_SITE_PACKAGES}"
-          PATTERN "*__pycache__" EXCLUDE
-          PATTERN "*.pyc" EXCLUDE
-          PATTERN "*.so" EXCLUDE
-          PATTERN "*.dylib" EXCLUDE)
-
+    PATTERN "*__pycache__" EXCLUDE
+    PATTERN "*.pyc" EXCLUDE
+    PATTERN "*.so" EXCLUDE
+    PATTERN "*.dylib" EXCLUDE
+)
 
 # builds the python bindings (must be after setting up the rpath)
-pybind11_add_module(pyRascal ${PY_BINDING_SRCS})
-target_link_libraries(pyRascal PRIVATE "${LIBRASCAL_NAME}")
-# Want to rename the output, so that the python module is called _rascal
-set_target_properties(pyRascal PROPERTIES OUTPUT_NAME _rascal)
-# Sets rule to install the bindings library in the rascal folder
-install(TARGETS pyRascal DESTINATION "${PYTHON_SITE_PACKAGES}/rascal/lib")
+# SYSTEM: use include_directory(SYSTEM)
+# NO_EXTRAS: do not setup LTO (link time optimization) and striping of the
+#            shared library 
+pybind11_add_module(_rascal ${BINDING_SRCS} SYSTEM NO_EXTRAS)
+target_compile_options(_rascal PRIVATE -Werror)
```

Comment:
is `-Wall`included additionally? I'm a little lost.

Reply:
-Wall is included in CMAKE_CXX_FLAGS in the root CMakeLists.txt, but this only sets -Werror for the _rascal target, not all of them (and in particular not on external libraries)